plugins/capsulesteps/ingress_routes/watcher.go

```diff
@@ -148,20 +148,22 @@ func onCertificateRequestUpdated(
 	return status
 }
 
-func onIngressUpdated(
+func (p *Plugin) onIngressUpdated(
 	obj client.Object,
 	events []*corev1.Event,
 	objectWatcher plugin.ObjectWatcher,
 ) *apipipeline.ObjectStatusInfo {
 	ingress := obj.(*netv1.Ingress)
 
-	objectWatcher.WatchSecondaryByLabels(labels.Set{
-		pipeline.LabelOwnedByCapsule: ingress.GetLabels()[pipeline.LabelOwnedByCapsule],
-	}.AsSelector(), &cmv1.Certificate{}, onCertificateUpdated)
+	if shouldCreateCertificateResource(p.config) {
```

Comment:
I think this should only happen on `!disableTLS` or whatever it's called

Reply:
what if `createCertificateResources` is false?

Reply:
that only means a clusterIssuer will create them for us... it's a bit of a weird flag :sweat_smile: